modules/aws/lb/main.tf

```diff
@@ -0,0 +1,301 @@
+###########################
+# Provider Configuration
+###########################
+terraform {
+  required_version = ">= 1.0.0"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.0.0"
+    }
+  }
+}
+
+###########################
+# Locals
+###########################
+
+locals {
+  is_application = var.load_balancer_type == "application"
+  is_network     = var.load_balancer_type == "network"
+}
+
+###########################################################
+# Load Balancer
+###########################################################
+
+resource "aws_lb" "load_balancer" {
+  # Common settings
+  name                       = var.name
+  internal                   = var.internal
+  load_balancer_type         = var.load_balancer_type
+  subnets                    = var.subnets
```

Comment:
Subnets and subnet_mappings are mutual exclusive, I think this needs some additional logic. Likely on each.

Reply:
not sure if I did this right. I added some lifecycle logic to the module, but also some validatio to the variable. Think I only need one and can just remove the one we don't want. 

Tried to add some additional logic for subnet mappings, but honestly, i got a little lost here. Maybe we can talk about what I added.